### PR TITLE
Added tests

### PR DESCRIPTION
**Pull request by alexsaveliev**
- added unit tests 
- added Travis CI tests
- updated README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,11 +14,38 @@ To try it in [Visual Studio Code](https://code.visualstudio.com), install the [v
 
 Run `node_modules/.bin/tsc --watch`.
 
+## Tests
+
+Run `node_modules/.bin/mocha build/test/*.js`
```
Comment (sqs) [inline README.md]: > Run `npm test` 

is more standard/simpler

Comment (sqs) [inline README.md]: or `yarn test` :)

File: src/test/test-utils.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            channel.serverConnection.start();
+            channel.clientConnection.start();
+
+            const params: vscode.InitializeParams = {
+                processId: null,
+                rootPath: 'file:///',
+                capabilities: null
+            }
+
+            channel.clientConnection.sendRequest(rt.InitializeRequest.type, params).then(function () {
```
Comment (sqs) [inline src/test/test-utils.ts]: This applies to the whole codebase, but why not use ES6 `() => ...` syntax? `function() { ... }` is much more verbose and has worse handling for `this` in most cases.

File: tsd.json [hunk 3]
```diff
@@ -7,6 +7,15 @@
   "installed": {
```
Comment (sqs) [inline tsd.json]: why this instead of `npm install --save-dev @types/whatever`?

File: src/test/test-utils.ts (hunk 2 above)
Comment (alexsaveliev) [inline src/test/test-utils.ts]: bad habit, sorry, will take into account in a separate PR :)

**Pull request merged**